src/aerospike_clj/key.clj

```diff
@@ -0,0 +1,37 @@
+(ns aerospike-clj.key
+  (:import [com.aerospike.client Key Value]
+           [com.aerospike.client.util ThreadLocalData]))
+
+(defprotocol AerospikeIndex
+  (create-key-method ^Key [this ^String as-namesapce ^String set-name]))
+
+(extend-protocol AerospikeIndex
+  (Class/forName "[B") 
+  (create-key-method ^Key [this ^String as-namesapce ^String set-name]
+    (when (< (ThreadLocalData/DefaultBufferSize) (+ 1 (alength #^bytes this) (.length set-name)))
+      (throw (Exception. (format "key is too long: %s..." this))))
+    (Key. ^String as-namesapce ^String set-name #^bytes this))
+  String 
+  (create-key-method ^Key [this ^String as-namesapce ^String set-name]
+    (when (< (ThreadLocalData/DefaultBufferSize) (+ 1 (.length this) (.length set-name)))
+      (throw (Exception. (format "key is too long: %s..." (subs this 0 40)))))
```

Comment:
Why should we not let the aerospike java lib handle long key exceptions?
[link](https://github.com/aerospike/aerospike-client-java/blob/master/client/src/com/aerospike/client/Key.java#L70)

Reply:
Because if you can save the trip to the db and fail locally it's better.

Reply:
it's actually to meant to prevent an exception when the local buffer is not big enough (there's no auto resize)

Reply:
the library throws an exception (not the server), but an unclear one. This explains better what happens and how to fix.